csrc/scheduler/vectorize_helper.cpp

```diff
@@ -229,148 +229,184 @@ void ContiguousInnerDimensionsMapper::distributePE(
   addProjectedExtent(merge_or_split->outer(), projected_outer_extent);
 }
 
-std::vector<IterDomain*> ContiguousInnerDimensionsMapper::projectIdToRoot(
-    TensorView* ref,
-    std::vector<IterDomain*> ids) {
-  auto transform_exprs = StmtSort::getExprs(
-      ref->fusion(),
-      {ref->getRFactorDomain().begin(), ref->getRFactorDomain().end()});
-
-  // Mapping from rfactor to root, reverse expressions
-  std::reverse(transform_exprs.begin(), transform_exprs.end());
+std::vector<IterDomain*> ContiguousInnerDimensionsMapper::projectId(
+    std::vector<IterDomain*> from,
+    const std::vector<IterDomain*>& to) {
+  if (from.empty()) {
+    return {};
+  }
 
-  for (auto* expr : transform_exprs) {
-    if (Split* split = dynamic_cast<Split*>(expr)) {
-      // Initialize state
-      auto find_outer_it = ids.begin();
-      auto outer_pos = ids.size();
-      auto find_inner_it = ids.begin();
-      auto inner_pos = ids.size();
-
-      // Removes all entries to the left of provided `it`, if `it` is not
-      // ids.begin(). Updates all state of finding outer and inner in the ids
-      // vector after erasing.
-      auto clear_left_of = [&find_outer_it,
-                            &outer_pos,
-                            &find_inner_it,
-                            &inner_pos,
-                            &ids,
-                            &split](decltype(find_outer_it) it) {
-        if (it != ids.begin()) {
-          ids.erase(ids.begin(), it);
-        }
+  // "combine" is a merge in forward propagation, and a split in backward
+  auto propagateCombine = [&](auto* merge_or_split) {
```

Comment:
This is just a copy-paste of the previous code with minor modification.

Reply:
Does this need to be a lambda? This function looks just too big to be a lambda function. There's even a nested lambda.

Reply:
It can be a private member function, but I don't feel it is easier to read than just using lambda, because what this function does is highly related to the context of `projectId`

Reply:
I'm not sure I agree with that. Having a standalone function would make it clearer what context is relevant as function parameters, instead of the automatic capture like `&`, which is convenient but implicit. Also it would promote you (at least for me) to write a proper function comment.

Reply:
I updated this function to use explicit capture and improvement the documentation. Do you still don't like it to be a lambda? The "context" I was referring to was about this:
![image](https://github.com/NVIDIA/Fuser/assets/1032377/fb722534-0b84-4935-9244-8dbfc6ca3724)
By using a lambda, we can just fold it inside `projectId` and read, and I think the folded version of the code is the most natural to read.

Reply:
Is it possible at the GH review page as well?

Reply:
Short answer: Unfortunately no, I have been hoping github to add this for a while
Long answer: You can click a `.` button on this page to enter a vscode editor called `github.dev`, and go to the whole file in that editor, then you will be able to get that. I personally never used this feature.